DrawMat/ViewModels/InteractionMode.cs

```diff
@@ -107,13 +110,34 @@ public void PointerMoved(MainViewModel vm, Point position)
 
     public void PointerReleased(MainViewModel vm, Point position)
     {
-        // TODO: find shapes inside
+        if (SelectionRect == null) return;
+        var rect = SelectionRect.Value;
+        SelectedShapes = vm.RootGroup.SearchChildren(new BoundingBox(rect.X, rect.Y, rect.Right, rect.Bottom));
         SelectionRect = null;
     }
 
+    public void PointerPressedRight(MainViewModel vm, Point position)
+    {
+        foreach (var child in SelectedShapes)
+        {
+            vm.RootGroup.Children.Remove(child);
+        }
+        SelectedShapes = new List<ShapeBase>();
+    }
+
     public IEnumerable<Control> GetVisuals()
     {
         var canvas = new Canvas();
+        if (SelectedShapes != null){
```

Comment:
This null check is now unnecessary. (Always check all uses of a variable when refactoring.)